pkg/cloudevents/transport/amqp/codec.go

```diff
@@ -196,17 +196,17 @@ func (c Codec) fromHeaders(h map[string]interface{}, event *cloudevents.Event) e
 	}
 	delete(h, prefix+"source")
 
-	if t, ok := h[prefix+"time"].(string); ok { // TODO: time can be empty
-		timestamp := types.ParseTimestamp(t)
-		if err := ec.SetTime(timestamp.Time); err != nil {
-			return err
+	if t, ok := h[prefix+"time"].(string); ok {
+		if timestamp := types.ParseTimestamp(t); timestamp != nil {
```

Comment:
Right now ParseTimestamp() doesn't return an error. It's debatable on what to do if there is a bad time being presented whether to fail entirely or, as in the case, silently drop. Perhaps open an issue?

Reply:
https://github.com/cloudevents/sdk-go/issues/188

Reply:
good points, I was optimizing for moving fast but it is true that this could catch someone off guard... I caught myself off guard. :D